saleor/graphql/account/mutations/base.py

```diff
@@ -149,8 +199,26 @@ def perform_mutation(cls, _root, info, **data):
                     )
                 }
             )
+        return user
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        email = data["email"]
+        redirect_url = data["redirect_url"]
+        channel_slug = data.get("channel_slug")
+        user = cls.clean_user(email, redirect_url)
+
+        if not user.is_staff:
+            channel_slug = clean_channel(channel_slug).slug
+        elif channel_slug is not None:
+            channel_slug = validate_channel(channel_slug).slug
```

Comment:
Shouldn't be:
```suggestion
            channel_slug = clean_channel(channel_slug).slug
        else:
            channel_slug = validate_channel(channel_slug).slug
```
as we have a check for empty `channel_slug` there?

Reply:
For customer we clean the channel and validate it. We will raise exception when we have multiple channels and input.channel is None.

But for staff user we will proceed with validation only when the input.channel is provided. Staff user are not related with the channel so there is no need to require it as an input. Adding the `else` statement, we will require a channel slug for each staff user.

Reply:
👍  My mistake